facebookresearch/LIGHT
# [Event Loggers][5/5] Add a basic script for loading back the events and graph states in an event log

## Overview
This PR depends on #37 

This branch adds a basic script which, when passed the name of a event log, will return a map of uuiid to worlds, and a list of (hash, timestamp, GraphEvent) tuples needed to completely replay what occurred from the perspective of the logger.

This is achieved by passing an --event-log via commandline with the script

## Testing
This PR's functionality is essentially copied and tested in most of #37.

Further testing can be accomplished as following:

1. Use `play_random_map.py` or `deploy/web/deploy.sh` to play a game

2. Find the event logs (default to `logs/light_event_dumps/...`

3. Choose a log to reload and run `python scripts/filtering/reconstruct_logs.py --event-log <event-log-file>`, and observe the output of reconstructed logs!

Example is below, where we reconstruct the log of a room with a graveyard keeper and our player agent

```

> python script/examples/play_random_map.py
...<OUTPUT TRIMMED>...
Your soul searches for a character to inhabit
...<OUTPUT TRIMMED>...
action> go south
A graveyard keeper said "i just saw the following: a master at arm arrived from the a path to the north"
You are in the Main entrance.
There are religious statues standing outside this temple.  The outside of the temple is fairly plain itself and imposing looking.  The floor is made out of stone with a bridge leading to the entrance.
There's an entrance here.
A graveyard keeper is here.
There's a path to the north.
A graveyard keeper smiles
action> say "Hi Mr. keeeper"
A graveyard keeper said "i just saw the following: a master at arm said "hi mr. keeeper""
A graveyard keeper smiles
action> go north
A graveyard keeper said "i just saw the following: a master at arm left towards "the a path to the north""
A graveyard keeper smiles
<...>

> python scripts/filtering/reconstruct_logs.py --event-log logs/light_event_dumps/room/main_entrance_7_b8cdc4cc-7346-44e6-a069-793711afc24b_events.log

{'03f435fd-5ec7-49c9-b5e0-0f41fc97fab5': <light.world.world.World object at 0x7fbb69020210>}
[('8779454292897', 'Thu Jul 16 08:05:14 2020', ArriveEvent(GraphAgent(master at arm_60), "the a path to the north")), ('8779391721085', 'Thu Jul 16 08:05:14 2020', LookEvent(GraphAgent(master at arm_60))), ('8779391705633', 'Thu Jul 16 08:05:14 2020', EmoteEvent(GraphAgent(graveyard keeper_65), "smile")), ('8779391682513', 'Thu Jul 16 08:05:23 2020', SayEvent(GraphAgent(master at arm_60), "Hi Mr. keeeper")), ('8779391705685', 'Thu Jul 16 08:05:23 2020', SayEvent(GraphAgent(graveyard keeper_65), "I just saw the following: A master at arm said "hi mr. keeeper"")), ('8779441683605', 'Thu Jul 16 08:05:23 2020', EmoteEvent(GraphAgent(graveyard keeper_65), "smile")), ('8779391705665', 'Thu Jul 16 08:05:33 2020', LeaveEvent(GraphAgent(master at arm_60), [GraphRoom(main street_6)]))]
```

Note, the graveyard keeper saying he saw the player go north was not logged because the player is no longer in the room!

## Next Steps

The next step after this PR is to construct a downstream pipeline which will consume these replayed events, and produce a dataset.
